### PR TITLE
[MI-2544] Done the review fixes of a PR #872

### PR DESCRIPTION
**Pull request by Nityanand13**
[PR: #872](https://github.com/mattermost/mattermost-plugin-jira/pull/872)

### CODE REVIEW

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/issue.go [hunk 1]
```diff
@@ -1141,13 +1141,11 @@ func (s *ConnectionSettings) ShouldReceiveNotification(role string) bool {
 }
 
 func (p *Plugin) fetchConnectedUserFromAccount(account map[string]string, instance Instance) (Client, *Connection, error) {
-	var mattermostUserID types.ID
-	var err error
+	accountKey := account["Name"]
 	if account["AccountID"] != "" {
```
Comment (Kshitij-Katiyar) [inline server/issue.go:1145]: why not do the same for account["AccountID"] like you did for account["Name"] means making a variable

Comment (Nityanand13) [reply server/issue.go:1145]: I think this improve more readability, also It is suggested in one of the review comments.

File: server/issue.go [hunk 2]
```diff
@@ -1165,28 +1165,26 @@ func (p *Plugin) fetchConnectedUserFromAccount(account map[string]string, instan
...
+	*accountInformation = append(*accountInformation, map[string]string{
+		jiraUserAccountID: accountID,
+		jiraUserName:      name,
+	})
+}
+
 func (wh *webhook) fetchConnectedUser(p *Plugin, instanceID types.ID) (Client, *Connection, error) {
 	var accountInformation []map[string]string
 
 	if wh.Issue.Fields != nil && wh.Issue.Fields.Creator != nil {
```
Comment (manojmalik20) [inline server/issue.go:1178]:
```suggestion
	if wh.Issue.Fields != nil {
	    if wh.Issue.Fields.Creator != nil {
```
This way, all the three conditions can be present inside the parent `if`.

File: server/settings.go [hunk 3]
```diff
@@ -39,7 +41,7 @@ func (p *Plugin) settingsNotifications(header *model.CommandArgs, instanceID, ma
 	}
 
 	var value bool
-	switch args[2] {
+	switch args[roleStatusIndex] {
```
[the remark this answers is not in the document]
Comment (Nityanand13) [reply server/settings.go:44]: but they asked to keep these magic numbers as a constant. https://github.com/mattermost/mattermost-plugin-jira/pull/872#discussion_r1060362020

[the remark this answers is not in the document]
Comment (manojmalik20) [reply server/settings.go:44]: Are these being used anywhere else outside this function?

[the remark this answers is not in the document]
Comment (Nityanand13) [reply server/settings.go:44]: For every command they are checking like this only args[0], args[1], etc. 

For this case, we are just checking that if `len(args) != 0 && args[0] ==  "notifications"` then we are calling this function and other than this I don't think that it is used anywhere else.

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**